### PR TITLE
8.1 is done

### PR DESCRIPTION
**Pull request by dmytro-hryhoruk**

### CODE REVIEW

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -1,50 +1,171 @@
...
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
+    public MyLinkedList() {
+    }
+
+    static class Node<T> {
+        T value;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MyLinkedList.java]: add access modifiers to all fields and methods, please

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 2]
```diff
@@ -1,50 +1,171 @@
...
+    private Node findByIndex(int index) {
+        checkIndex(index);
+        Node value;
+        if (index < (size >> 1)) {
+            value = head;
+            for (int i = 0; i < index; i++) {
+                value = value.next;
+            }
+            return value;
+        } else {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MyLinkedList.java]: no need to use else after return statement, isn't it?

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 3]
```diff
@@ -1,50 +1,171 @@
...
+                value = value.prev;
+            }
+            return value;
+        }
+    }
+
+    private int findByValue(T value) {
+        Node result = head;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(value, result.value)) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MyLinkedList.java]: try to solve without Objects class

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 4]
```diff
@@ -3,48 +3,167 @@
...
+        return (T) first.value;
+    }
+
+    private T removeFromInside(Node element, int index) {
+        Node prevElement = findByIndex(index - 1);
+        Node nextElement = findByIndex(index + 1);
+        prevElement.next = nextElement;
+        nextElement.prev = prevElement;
+        size--;
+        return (T) element.value;
```
[the remark this answers is not in the document]
Comment (boroda4436) [reply src/main/java/core/basesyntax/MyLinkedList.java]: Let's leave it as is

**Pull request closed without merging**